prov/util/src/util_mr_cache.c

```diff
@@ -393,12 +393,21 @@ struct ofi_mr_entry *ofi_mr_cache_find(struct ofi_mr_cache *cache,
 {
 	struct ofi_mr_info info;
 	struct ofi_mr_entry *entry;
+	struct ofi_mem_monitor *monitor;
+	bool flush_lru;
 
 	assert(attr->iov_count == 1);
 	FI_DBG(cache->prov, FI_LOG_MR, "find %p (len: %zu)\n",
 	       attr->mr_iov->iov_base, attr->mr_iov->iov_len);
 
 	pthread_mutex_lock(&mm_lock);
+	flush_lru = ofi_mr_cache_full(cache);
+	if (flush_lru || !dlist_empty(&cache->dead_region_list)) {
+		pthread_mutex_unlock(&mm_lock);
+		ofi_mr_cache_flush(cache, flush_lru);
+		pthread_mutex_lock(&mm_lock);
+	}
```

Comment:
I'm not following the reasoning for this change.  find() doesn't insert new entries into the cache, so why the need to flush?  It would have the impact of slowing down the find().  That may be okay or desirable, but I'm not sure.

Reply:
That's a fair point. Maybe this change to the following to having some cleanup of evicted entries?

```
if (!dlist_empty(&cache->dead_region_list)) {
    pthread_mutex_unlock(&mm_lock);
    ofi_mr_cache_flush(cache, flse);
    pthread_mutex_lock(&mm_lock);
}
```

Reply:
I'm okay with flushing dead entries, but can you move this part of the change to its own commit?  If it shows some problem with performance, it'll be easier to revert if needed.

Reply:
Can do.